firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDataMigrationManager.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import com.google.firebase.firestore.auth.User;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.mutation.MutationBatch;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/** Manages SQLite data migration required by SDK version upgrades. */
+public class SQLiteDataMigrationManager implements DataMigrationManager {
+  private final SQLitePersistence db;
+  private final Set<SQLitePersistence.DataMigration> migrations;
+
+  /**
+   * Creates a new data migration manager.
+   *
+   * @param persistence The underlying SQLite Persistence to use for data migrations.
+   * @param migrations The set of migrates needed to run.
+   */
+  public SQLiteDataMigrationManager(
```

Comment:
I have a couple high level questions:

How do we store the fact that the migrations have already run? How do we make sure they run at least once? What happens if the user closes the app right after the schema upgrade?

Reply:
Good point. I introduced a new table for this..